OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,124 +1,158 @@
-                               pt->GetValues().coeff(f, 1),
-                               pt->GetValues().coeff(f, 2)};
-    }
-
-    
-marker_table.appendRow(0 + f * time_step, row);
+Rot = -90;
+rotationMatrix = [1,0,0;0,cosd(Rot),-(sin(Rot*pi/180));0,sin(Rot*pi/180),cosd(Rot)];
 
+%% Rotate marker data
+% make a clean copuy to alter
+markers_rotated = markers();
 
-for iMarker = 0 : length(markerdata) - 1
+for iMarker = 0 : nMarkers - 1
 
     % get the column data for the marker
-    marker = markers.updDependentColumn(markerlabels(iMarker+1) );
+    marker = markers_rotated.updDependentColumnAtIndex(iMarker);
 
     % go through each element of the table column, rotate the Vec3, and write
     % back to the column.
-    for iRow = 0 : markers.getNumRows - 1
+    for iRow = 0 : markers_rotated.getNumRows - 1
         % get Matlab vector marker position
         vectorData = [marker.getElt(0,iRow).get(0)...
                       marker.getElt(0,iRow).get(1)...
                       marker.getElt(0,iRow).get(2)];
 
         % rotate the marker data
-        rotatedData = [rotationMatrix'*vectorData']';
+        rotatedData = [rotationMatrix*vectorData']';
 
         % Write the rotated data back to the Vec3TimesSeriesTable
-        Vec3(rotatedData(1),rotatedData(2),rotatedData(3))
-        
-        methodsview(new_table)
-                 
-          create an empty timesseriestable and use append every row Vec3
+        elem = Vec3(rotatedData(1),rotatedData(2),rotatedData(3));
+        % set the value of the element
+        marker.set(iRow, elem);
     end
 end
 
-%% Print the rotated markers to trc file
-trcfileadapter = TRCFileAdapter();
-trcfileadapter.write(markers,'test_walking_rotated.trc');
-
 %% Rotate Force data
+% make a clean copy to alter
+forces_rotated = forces();
 
-for iForces = 0 : length(forcesdata) - 1
+for iForce = 0 : nForces - 1
 
     % get the column data for the marker
-    force = markers.updDependentColumn(forcelabels(iForce+1) );
+    force = forces_rotated.updDependentColumnAtIndex( iForce   );
 
     % go through each element of the table column, rotate the Vec3, and write
     % back to the column.
-    for iRow = 0 : forces.getNumRows - 1
+    for iRow = 0 : nForceFrames - 1
         % get Matlab vector marker position
         vectorData = [force.getElt(0,iRow).get(0)...
                       force.getElt(0,iRow).get(1)...
                       force.getElt(0,iRow).get(2)];
 
         % rotate the marker data
-        rotatedData = [rotationMatrix'*vectorData']';
+        rotatedData = [rotationMatrix*vectorData']';
 
         % Write the rotated data back to the Vec3TimesSeriesTable
-        % THE BELOW LINES DO NOT WORK AND THERE IS CURRENTLY NO FIX
-        % force.updElt(0,iRow).set(0, rotatedData)
-             
+        elem = Vec3(rotatedData(1),rotatedData(2),rotatedData(3));
+        % set the value of the element
+        force.set(iRow, elem);
```

Comment:
lines 117-129 would seem to be very time consuming, because `rotatedData` and `elem` are allocated over and over again and not because of any math. 

Since you are converting to another `TimeSeriesTableVec3` (and not a Matlab data structure) I would consider doing the multiplies on the OpenSim/SimTK  side. For example, we can do this:
```cpp
        double ang = SimTK::Pi / 3;
        Rotation r(ang, SimTK::XAxis);
        SimTK::Vector_<SimTK::Vec3> packedV(5, Vec3(2));  // a packed Vector of Vec3s
        packedV.dump("packedV");
        auto rotatedV = r*packedV;                        // rotate all Vec3s in packedV at once 
        rotatedV.dump("rotateV");
 ```

I think this would speed things up a lot.